### PR TITLE
DS-3154 fix javadoc

### PR DESCRIPTION
**Pull request by helix84**
https://jira.duraspace.org/browse/DS-3154

This actually fixes the bulk of the JavaDoc errors. Only a few remain.

### CODE REVIEW

**Comment by tdonohue** [comment]
+1 via code review. This is almost all comment fixes, with a few small spelling errors fixed in log statements, etc.

File: dspace-api/src/main/java/org/dspace/core/AbstractHibernateDAO.java [hunk 1]
```diff
@@ -94,8 +94,8 @@ public T findByID(Context context, Class clazz, int id) throws SQLException {
      * Execute a JPA Criteria query and return a collection of results.
      *
      * @param context
-     * @param query
-     * @return
+     * @param query JPQL query string
+     * @return list of DAOs specified by the query string
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/core/AbstractHibernateDAO.java]: DSOs, perhaps?

**Comment by hardyoyo** [comment]
+1 by inspection

File: dspace-api/src/main/java/org/dspace/identifier/doi/DataCiteConnector.java [hunk 2]
```diff
@@ -774,7 +774,7 @@ protected DataCiteResponse sendHttpRequest(HttpUriRequest req, String doi)
                 content = EntityUtils.toString(entity, "UTF-8");
             }
 
-            /* While debugging it can be useful to see whitch requests are send:
+            /* While debugging it can be useful to see which requests are sent:
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/identifier/doi/DataCiteConnector.java]: send -> sent

File: dspace-api/src/main/java/org/dspace/rdf/storage/RDFStorage.java [hunk 3]
```diff
@@ -18,21 +18,21 @@
 public interface RDFStorage {
     /**
      * Don't use this method directly, use 
-     * {@link org.dspace.rdf.RDFizer#convert(org.dspace.core.Context, 
-     * org.dspace.content.DSpaceObject) RDFizer.convert(...)} to convert and store DSpaceObjets.
+     * {@link org.dspace.rdf.RDFUtil#convert(org.dspace.core.Context, org.dspace.content.DSpaceObject) RDFizer.convert(...)}
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/rdf/storage/RDFStorage.java]: Link and text don't match.

**Comment by mwoodiupui** [comment]
+1 by inspection.  Much improvement here.

**Comment by tdonohue** [comment]
Merging as this is at +2. I'll add a note to make the few (minor) comment fixes in a followup PR.  @helix84 noted that there seems to be few javadocs issues left, so I'll see if I can quickly resolve those in a followup.

**Pull request merged**